bundles/org.dataflowanalysis.analysis.converter/src/org/dataflowanalysis/analysis/converter/PalladioConverter.java

```diff
@@ -0,0 +1,199 @@
+        Node dfdNode = createCorrespondingDFDNode(pcmASE);
+        dfdNodeMap.put(pcmASE.getElement(), dfdNode);
+        return dfdNode;
+    }
+
+    private Node createCorrespondingDFDNode(AbstractPCMActionSequenceElement<? extends Entity> pcmASE) {
+        Node node;
+
+        if (pcmASE instanceof UserActionSequenceElement<?>) {
+            node = dataflowdiagramFactory.eINSTANCE.createExternal();
+        } else if (pcmASE instanceof SEFFActionSequenceElement<?>) {
+            node = dataflowdiagramFactory.eINSTANCE.createProcess();
+        } else {
+            logger.error("Unregcognized palladio element");
+            return null;
+        }
+
+        Behaviour behaviour = datadictionaryFactory.eINSTANCE.createBehaviour();
+        node.setEntityName(pcmASE.getElement().getEntityName());
+        node.setId(pcmASE.getElement().getId());
+        node.setBehaviour(behaviour);
+        dataDictionary.getBehaviour().add(behaviour);
+        dataFlowDiagram.getNodes().add(node);
+        return node;
+    }
+
+    private void addNodeCharacteristicsToNode(Node node, List<CharacteristicValue> charValues) {
+        for (CharacteristicValue charValue : charValues) {
+            Label label = getOrCreateDFDLabel(charValue);
+            if (!node.getProperties().contains(label)) {
+                node.getProperties().add(label);
+            }
+        }
+    }
+
+    private Label getOrCreateDFDLabel(CharacteristicValue charValue) {
+        LabelType type = dataDictionary.getLabelTypes().stream().filter(f -> f.getEntityName().equals(charValue.getTypeName())).findFirst()
+                .orElse(createLabelType(charValue));
+
+        Label label = type.getLabel().stream().filter(f -> f.getEntityName().equals(charValue.getValueName())).findFirst()
+                .orElse(createLabel(charValue, type));
+
+        return label;
+    }
+
+    private Label createLabel(CharacteristicValue charValue, LabelType type) {
+        Label label = datadictionaryFactory.eINSTANCE.createLabel();
+        label.setEntityName(charValue.getValueName());
+        type.getLabel().add(label);
+        return label;
+    }
+
+    private LabelType createLabelType(CharacteristicValue charValue) {
+        LabelType type = datadictionaryFactory.eINSTANCE.createLabelType();
+        type.setEntityName(charValue.getTypeName());
+        this.dataDictionary.getLabelTypes().add(type);
+        return type;
+    }
+
+    /**
+     * Converts a Palladio model into a DataFlowDiagramAndDictionary object.
+     * @param inputModel Name of the model folder.
+     * @patam inputFile Name of the model file.
+     * @param modelLocation Location of the model folder.
+     * @return DataFlowDiagramAndDictionary object representing the converted Palladio model.
+     */
+    public DataFlowDiagramAndDictionary assToDFD(String inputModel, String inputFile, String modelLocation) {
```

Comment:
Please sort methods in the classes so that the entry method, i.e., the public ones is on top. A good class reads like a newspaper (quote from the clean code book)